### PR TITLE
Providing API version header from server to caller when available.

### PR DESCRIPTION
**Pull request by OrangeAndGreen**
Attempts to safely retrieve the x-api-current-version header from API response payloads so calling code can do version checking.
https://dimagi.atlassian.net/browse/CCCT-239

cross-request: https://github.com/dimagi/commcare-core/pull/1409

### CODE REVIEW

File: app/src/org/commcare/tasks/ModernHttpTask.java [hunk 1]
```diff
@@ -107,4 +114,9 @@ protected void deliverError(HttpResponseProcessor httpResponseProcessor,
     public InputStream getResponseStream() {
         return responseDataStream;
     }
+
+    @Override
+    public String getApiVersion() {
+        return apiVersion;
```
Comment (shubham1g5) [inline app/src/org/commcare/tasks/ModernHttpTask.java:120]: how about `return requester.getApiVersion()` ? 

Also is this API version connect only or will also be added to non connect server calls ?

Comment (OrangeAndGreen) [reply app/src/org/commcare/tasks/ModernHttpTask.java:120]: Ah, excellent suggestion, I'll do that!

Currently both the ConnectId and Connect servers are returning this header. I'm not sure about any plans for HQ to use it.

**Comment by OrangeAndGreen** [comment]
@damagatchi Retest this please

**Review by shubham1g5 — APPROVED** [review]

**Comment by OrangeAndGreen** [comment]
@damagatchi retest this please

**Comment by OrangeAndGreen** [comment]
@damagatchi retest this please

**Comment by OrangeAndGreen** [comment]
@damagatchi retest this please

**Comment by OrangeAndGreen** [comment]
@damagatchi retest this please

**Comment by shubham1g5** [comment]
@damagatchi retest this please

Since the core PR is merged without these changes we are now breaking [the master build](https://jenkins.dimagi.com/job/commcare-android/). Our integration tests are failing currently due to Password provisioning, hence I am running without integration tests to be able to merge this.

**Comment by OrangeAndGreen** [comment]
Ah, yes, thanks for that. I'd considered skipping the integration tests as well since the failures didn't seem related to these changes, but wasn't sure whether that would be advised.

**Pull request merged**